components/dialog/dialog.js

```diff
@@ -120,16 +143,29 @@ export class Dialog extends HTMLElement {
      * @param title
      * @returns {Promise<void>}
      */
-    async #setHeader(header, title) {
-        const headerElement = this.shadowRoot.querySelector("#header");
+    async #setHeader(header, options) {
+        const headerElement = this.shadowRoot.querySelector("header");
+        if (options?.severity != null) {
+            headerElement.dataset.severity = options.severity;
+        }
 
-        if (title != null) {
-            headerElement.querySelector("#headerText").textContent = options.title;
+        if (options?.showResize === false) {
+            headerElement.dataset.showResize = "false";
         }
 
         if (header != null) {
-            headerElement.insertBefore(options.header, headerElement.firstElementChild);
+            headerElement.innerHTML = "";
```

Comment:
this is two write operations to update header. is there not better way to do this

Reply:
Used replaceChildren instead